### ISSUE TITLE
memneq.hpp issue

### ISSUE THREAD

**Issue by berserker** [OpenVPN/openvpn3#38]
I managed to get the last master branch up & running on my Android device.
During development I got some random crashes while connecting to a server and I found an issue in this code of openvpn/common/memneq.hpp:
```
inline bool memneq_deref_ok(const void *a, const void *b) 
{
      return memneq_unaligned_ok || (is_aligned<const void *, memneq_t>(a)|is_aligned<const void *, memneq_t>(b));
}
```
that should be changed to:
```
inline bool memneq_deref_ok(const void *a, const void *b) 
{
      return memneq_unaligned_ok || (is_aligned<const void *, memneq_t>(a) && is_aligned<const void *, memneq_t>(b));
}
```
(please note the **&&** operator instead of the previous **|**)

The above change fixed all of my connection problems and looks coherent to me with the "old" code changed in [this commit](https://github.com/OpenVPN/openvpn3/commit/4e39a533e9d3513507d5a412956bf87a6c167498)

Looking for your feedback, thanks

**Comment by schwabe** [comment]
This is under our investigation and looks like the arm32 code generated by clang does not work anymore. We will push out a fix/workaround for this soon.

**Comment by dsommers** [comment]
I believe this should be fixed by commit 30ea53cb92ab7650b03092175041fdb10ab7f445 and de7c672ee7cc3061bcea0c158feb5ebaf9f84e5b, which was just pushed out.  Please reopen if this is still an issue.
